src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -125,6 +130,21 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
         }
 
         codeDocument.SetDocumentIntermediateNode(document);
+
+        static bool TryRemoveGlobalPrefixFromDefaultUsing(in UsingReference usingReference, out UsingReference trimmed)
+        {
+            const string globalPrefix = "global::";
+            if (usingReference.Source is { FilePath: null } && // the default import has null file path
+                usingReference.Namespace.StartsWith(globalPrefix, StringComparison.Ordinal))
+            {
+                trimmed = new UsingReference(
```

Comment:
This feels like it could be a major source of allocations. The list of default imports is pretty small, can we just maintain a map by hand?

Reply:
Removed the allocations, not sure if you think a map would be beneficial still?